src/screens/CreatePost/index.js

```diff
@@ -435,54 +435,56 @@ const CreatePost = () => {
       return true;
     }
 
-    const topicsToPost = _.union(initialTopic, listTopic);
-    const data = {
-      message,
-      topics: topicsToPost,
-      verb: isPollShown ? 'poll' : 'tweet',
-      feedGroup: 'main_feed',
-      privacy: listPrivacy[0].key,
-      anonimity: typeUser,
-      location: renderLocationString(geoList[geoSelect]),
-      location_id: locationId,
-      duration_feed: postExpired[expiredSelect].value,
-      images_url: dataImage,
-      tagUsers: checkTaggingUser(),
-      is_photo_uploaded: true
-    };
-
-    if (isPollShown) {
-      data.polls = getReducedPoll();
-      data.pollsduration = selectedTime;
-      data.multiplechoice = isPollMultipleChoice;
-    }
-
-    if (typeUser) {
-      data.anon_user_info = {
-        color_name: anonUserInfo?.colorName,
-        color_code: anonUserInfo?.colorCode,
-        emoji_name: anonUserInfo?.emojiName,
-        emoji_code: anonUserInfo?.emojiCode
-      };
-    }
-
-    setDurationId(JSON.stringify(expiredSelect));
-    if (!isInCreatePostTopicScreen) {
-      setLocationId(JSON.stringify(geoSelect));
-      setPrivacyId(JSON.stringify(0));
-    }
-
     const maxRetries = 5;
-
     const attemptPost = async (retryCount) => {
       try {
+        const topicsToPost = _.union(initialTopic, listTopic);
+        const data = {
+          message,
+          topics: topicsToPost,
+          verb: isPollShown ? 'poll' : 'tweet',
+          feedGroup: 'main_feed',
+          privacy: listPrivacy[0].key,
+          anonimity: typeUser,
+          location: renderLocationString(geoList[geoSelect]),
+          location_id: locationId,
+          duration_feed: postExpired[expiredSelect].value,
+          images_url: dataImage,
+          tagUsers: checkTaggingUser(),
+          is_photo_uploaded: true
+        };
+
+        if (isPollShown) {
+          data.polls = getReducedPoll();
+          data.pollsduration = selectedTime;
+          data.multiplechoice = isPollMultipleChoice;
+        }
+
+        if (typeUser) {
+          data.anon_user_info = {
+            color_name: anonUserInfo?.colorName,
+            color_code: anonUserInfo?.colorCode,
+            emoji_name: anonUserInfo?.emojiName,
+            emoji_code: anonUserInfo?.emojiCode
+          };
+        }
+
+        setDurationId(JSON.stringify(expiredSelect));
+        if (!isInCreatePostTopicScreen) {
+          setLocationId(JSON.stringify(geoSelect));
+          setPrivacyId(JSON.stringify(0));
+        }
+
         const post = await createPost(data);
+        if (params.onRefresh && typeof params.onRefresh === 'function') {
+          params.onRefresh();
+        }
+        setLoadingPost(false);
         if (post.code === 200) {
           showMessage({
             message: StringConstant.createPostDone,
             type: 'success'
           });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `attemptPost` function has been refactored to include the creation of the `data` object and the setting of various states within it. This is a good change as it encapsulates all related logic within the function, improving readability and maintainability. However, there's a potential issue with the error handling. If the `createPost` function throws an exception other than when `post.code !== 200`, the `setLoadingPost(false)` statement won't be executed, leaving the application in a loading state indefinitely. To fix this, you should move `setLoadingPost(false)` into a `finally` block to ensure it always runs.

```diff
         const post = await createPost(data);
        if (params.onRefresh && typeof params.onRefresh === 'function') {
          params.onRefresh();
        }
-        setLoadingPost(false);
         if (post.code === 200) {
           showMessage({
             message: StringConstant.createPostDone,
             type: 'success'
           });
 
           if (isInCreatePostTopicScreen) {
             navigateToTopicPage();
+          }
+        } finally {
+          setLoadingPost(false);
         }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->